forest/tree/tree.go

```diff
@@ -0,0 +1,58 @@
+package tree
+
+import (
+	cache "github.com/treeverse/lakefs/forest/cache_map"
+	gr "github.com/treeverse/lakefs/graveler"
+	"github.com/treeverse/lakefs/graveler/committed/sstable"
+)
+
+var minimalKey = gr.Key(nil)
+
+type treePartType struct {
+	PartName sstable.ID `json:"part_name"`
+	MaxKey   gr.Key     `json:"max_path"`
+}
+type TreeType struct {
+	treeSlice []treePartType
+}
+
+type treesRepo struct {
+	treesMap   cache.CacheMap
+	partManger sstable.Manager
+}
+
+const (
+	CacheMapSize     = 1000
+	CacheTrimSize    = 100
+	InitialWeight    = 64
+	AdditionalWeight = 16
+	TrimFactor       = 1
+)
+
+func InitTreesRepository(manager sstable.Manager) TreeRepo {
+	treesRepository := &treesRepo{
+		treesMap:   cache.NewCacheMap(CacheMapSize, CacheTrimSize, InitialWeight, AdditionalWeight, TrimFactor),
+		partManger: manager,
+	}
+	return treesRepository
+}
+
+func (t *treesRepo) GetPartManger() sstable.Manager {
+	return t.partManger
+}
+
+type TreeRepo interface {
+	NewTreeWriter(splitFactor int, closeAsync sstable.BatchWriterCloser) TreeWriter
+	NewScannerFromID(treeID gr.TreeID, start gr.Key) (gr.ValueIterator, error)
+	NewScannerFromTreeParts(treeSlice TreeType, start gr.Key) (gr.ValueIterator, error)
+	GetPartManger() sstable.Manager
+}
+
+type TreeWriter interface {
+	HasOpenWriter() bool
```

Comment:
It's not really clear to me how the writer works.
Are `HasOpenWriter`, `ForceCloseCurrentPart` and `IsSplitKey` part of the interface, or are they implementation details?

Reply:
HasOpenWriter and ForceClose are interfaces needed by Apply. 
I am not sure they will be needed by others. If Merge is implemented without apply, It will probably need it.
IsSplitKey determines if a key should terminate a part. is probably redundant. So unless someone really wants it - it will be removed